src/INode.java

```diff
@@ -229,6 +242,17 @@ private ASTValue evalDefineProperty(WordsEnvironment environment) {
 		throw new AssertionError("Not yet implemented");	
 	}
 
+	private ASTValue evalDivide(WordsEnvironment environment) throws WordsRuntimeException {
+		ASTValue lhs = children.get(0).eval(environment).tryCoerceTo(AST.ValueType.NUM);
+		ASTValue rhs = children.get(1).eval(environment).tryCoerceTo(AST.ValueType.NUM);
+
+		if ((lhs.type != AST.ValueType.NUM) || (rhs.type != AST.ValueType.NUM)) {
+			throw new WordsArithmeticException(lhs.type.toString(), rhs.type.toString());
```

Comment:
It will be confusing to the user if the lhs is a number and rhs is a string, (so the condition is true), but the exception thrown and ultimately displayed to the user says number (since lhs is a number).  Perhaps it's better to check each operand independently for lhs and rhs, and throw the exception with lhs or rhs type, so that the type shown to the user is actually the one that caused the problem.